packages/cli/src/WorkflowRunner.ts

```diff
@@ -191,6 +209,89 @@ export class WorkflowRunner {
 		return executionId;
 	}
 
+	async runBull(data: IWorkflowExecutionDataProcess, loadStaticData?: boolean, realtime?: boolean): Promise<string> {
+
+		// TODO: If "loadStaticData" is set to true it has to load data new on worker
+
+		// Register the active execution
+		const executionId = this.activeExecutions.add(data, undefined);
+
+		const jobData: IBullJobData = {
+			destinationNode: data.destinationNode,
+			executionId,
+			executionMode: data.executionMode,
+			executionData: data.executionData,
+			loadStaticData: !!loadStaticData,
+			retryOf: data.retryOf,
+			runData: data.runData,
+			startNodes: data.startNodes,
+			workflowData: data.workflowData,
+		};
+
+		let priority = 100;
+		if (realtime === true) {
+			// Jobs which require a direct response get a higher priority
+			priority = 50;
+		}
+		// TODO: For realtime jobs should probably also not do retry or not retry if they are older than x seconds.
+		//       Check if they get retried by default and how often.
+		const jobOptions = {
+			priority,
+			removeOnComplete: true,
+			removeOnFail: true,
+		};
+		const job = await this.jobQueue.add(jobData, jobOptions);
+		console.log('Started with ID: ' + job.id.toString());
+
+		const hooks = WorkflowExecuteAdditionalData.getWorkflowHooksWorkerMain(data.executionMode, executionId, data.workflowData, { retryOf: data.retryOf ? data.retryOf.toString() : undefined });
+
+		// Normally also workflow should be supplied here but as it only used for sending
+		// data to editor-UI is not needed.
+		hooks.executeHookFunctions('workflowExecuteBefore', []);
+
+		const workflowExecution: PCancelable<IRun> = new PCancelable(async (resolve, reject, onCancel) => {
+			onCancel.shouldReject = false;
+			onCancel(async () => {
+				if (await job.isActive()) {
+					// Job is already running so tell it to stop
+					await job.progress(-1);
+				} else {
+					// Job did not get started yet so remove from queue
+					await job.remove();
```

Comment:
I am not sure how this works here, but maybe we should add a `try...catch` block as the job may have started in the meantime, right? The time in between `isActive()` and `remove()`.

This might fail in the end in a high concurrency environment.

Reply:
Yes, there could maybe also other reasons why it could fail. So adding `try...catch` and then reporting back to UI if it fails should be added as you said.